src/horizon_client.rs

```diff
@@ -1911,41 +1962,135 @@ mod tests {
+        assert!(effects_for_liquidity_pools_response.clone().is_ok());
+
+        assert_eq!(
+            effects_for_liquidity_pools_response
+                .clone()
+                .unwrap()
+                ._embedded()
+                .records()[0]
+                .id(),
+            ID
+        );
+
+        assert_eq!(
+            effects_for_liquidity_pools_response
+                .clone()
+                .unwrap()
+                ._embedded()
+                .records()[0]
+                .paging_token(),
+            PAGING_TOKEN
+        );
 
-        assert!(_effects_for_ledger_response.clone().is_ok());
+        assert_eq!(
+            effects_for_liquidity_pools_response
+                .clone()
+                .unwrap()
+                ._embedded()
+                .records()[0]
+                .account(),
+            ACCOUNT
+        );
 
         assert_eq!(
-            _effects_for_ledger_response
+            effects_for_liquidity_pools_response
                 .clone()
                 .unwrap()
                 ._embedded()
                 .records()[0]
-                .id,
-            "0000000536870916097-0000000001"
+                .effect_type(),
+            RECORD_TYPE
         );
 
         assert_eq!(
-            _effects_for_ledger_response
+            effects_for_liquidity_pools_response
                 .clone()
                 .unwrap()
                 ._embedded()
-                .records()[1]
-                .effect_type,
-            "account_debited"
+                .records()[0]
+                .type_i(),
+            &TYPE_I
+        );
+
+        assert_eq!(
+            effects_for_liquidity_pools_response
+                .clone()
+                .unwrap()
+                ._embedded()
+                .records()[0]
+                .created_at(),
+            CREATED_AT
+        );
+
+        assert_eq!(
+            effects_for_liquidity_pools_response
+                .clone()
+                .unwrap()
+                ._embedded()
+                .records()[0]
+                .starting_balance()
+                .as_ref()
+                .unwrap(),
+            &STARTING_BALANCE
         );
     }
 }
+
+#[tokio::test]
+async fn test_get_effects_for_ledger() {
+    // found by trial and error in the Stellar laboratory
+    let ledger_sequence = 125;
+
+    let horizon_client =
+        HorizonClient::new("https://horizon-testnet.stellar.org".to_string()).unwrap();
+
+    let effects_for_ledger_request = EffectsForLedgerRequest::new().set_sequence(ledger_sequence);
+    let _effects_for_ledger_response = horizon_client
+        .get_effects_for_ledger(&effects_for_ledger_request)
+        .await;
+
+    println!("{:#?}", _effects_for_ledger_response);
+
+    assert!(_effects_for_ledger_response.clone().is_ok());
+
+    assert_eq!(
+        _effects_for_ledger_response
+            .clone()
+            .unwrap()
+            ._embedded()
+            .records()[0]
```

Comment:
Ik zou het record 1 keer ophalen en hieronder ook weer hergebruiken.

Reply:
lijkt me handig om dit ook bij meerdere endpoints in een keer te doen bij dezelfde refactor

Reply:
Ik zie m toch liever nu al mee in deze pr, opdat we het niet vergeten.